testrtc.py

```diff
@@ -21,7 +27,42 @@ def get(self, size_kbytes):
     self.response.headers.add_header("Access-Control-Allow-Origin", "*")
     self.response.headers['Content-Type'] = 'application/octet-stream'
     self.response.out.write(random_file[0: int(size_kbytes)*1000])
-    
+
+class NewReportHandler(blobstore_handlers.BlobstoreUploadHandler):
+  # Generate and return blobstore upload URL.
+  def head(self):
+    upload_url = blobstore.create_upload_url('/report/new')
+    self.response.headers['Response-Text'] = upload_url
+
+  # On uploaded file to the blobstore.
+  def post(self):
+    origin = self.request.headers['Origin']
+    fileName = self.request.headers['X-File-Name']
+    upload_files = self.get_uploads(field_name=fileName)
+    blob_info = upload_files[0]
+    self.response.headers['Response-Text'] = origin + '/report/%s' % blob_info.key()
+
+class ReportHandler(blobstore_handlers.BlobstoreDownloadHandler):
+  def get(self, resource):
+    resource = str(urllib.unquote(resource))
+    blob_info = blobstore.BlobInfo.get(resource)
+    self.send_blob(blob_info)
+
+# Remove log files equal or older than retentionPeriod.
+class CleanBlobstore(blobstore_handlers.BlobstoreUploadHandler):
+  def get(self):
+    retentionPeriod = datetime.datetime.now() - datetime.timedelta(days=90)
+    gqlQuery = blobstore.BlobInfo.gql("WHERE creation <= :date",
+        date = datetime.datetime.now() <= retentionPeriod)
+    logFiles = gqlQuery.run()
+    for log in logFiles:
+      # Make sure to only delete log reports from the blobstore.
+      if re.match(r'testrtc-.*.log', log.filename):
+        blobstore.BlobInfo.get(log.key()).delete()
+
 app = webapp2.WSGIApplication([
+    ('/report/new', NewReportHandler),
+    ('/report/([^/]+)', ReportHandler),
```

Comment:
Sorry for asking to change this one more time... but can we have '/report/([^/]+)/'  (notice an extra /) in the end.